### ISSUE TITLE
non-substantial speedups of sparse OPS

### ISSUE THREAD

**Issue by nikrao** [eric-haibin-lin/mxnet#136]
For bugs or installation issues, please provide the following information.
The more information you provide, the more likely people will be able to help you.

## Environment info
Operating System:

Compiler:

Package used (Python/R/Scala/Julia): Python

Or if installed from source:

MXNet commit hash (`git rev-parse HEAD`): -b sparse

[profile_output.json.zip](https://github.com/eric-haibin-lin/mxnet/files/1189190/profile_output.json.zip)

I am using the sparse ops to train a 2-layer fully connected NN. When I compare with my original dense implementation, the speedup is not significant. (on;y about 33%). When I run profiler, it seems most of the time is taken in the dense, fully connected layer backward computations (json attached). 

The custom OPS that were built for the specific loss function being used are reasonably fast. 

The code to build the network is below:
depth = 2
widths = 2048
nonlins = "relu"

`def make_network(depth,widths,nonlins,out_dim,dropouts,beta,rho,dim_in,label='softmax_label'):`
`    data = mx.sym.Variable('data', stype='csr')`
`    labs = mx.sym.Variable(label)`
`    norm = mx.initializer.Normal(sigma=0.01)`
`    v = mx.symbol.Variable("v", shape=(dim_in, widths[0]), init=norm,stype='row_sparse')`
`    net = mx.sym.dot(data,v)`
`    net = mx.sym.Activation(data=net,name='ac1',act_type=nonlins[0])`
`    net = mx.sym.IdentityAttachKLSparseReg(data=net,sparseness_target=rho,penalty=beta,name='kl1')`
`    net = mx.sym.Dropout(data=net,name='dl1',p=dropouts[0])`
`	# remaining hidden layers`
`    for layer in xrange(2,depth+1):`
`    	net = mx.sym.FullyConnected(data=net,name='fc'+str(layer),num_hidden=widths[layer-1])`
`        net = mx.sym.Activation(data=net,name='ac'+str(layer),act_type=nonlins[layer-1])`
`        net = mx.sym.IdentityAttachKLSparseReg(data=net,sparseness_target=rho,penalty=beta,name='kl'+str(layer))`
`        net = mx.sym.Dropout(data=net,name='dl'+str(layer),p=dropouts[layer-1])`
`    # output layer`
`    net = mx.sym.FullyConnected(data=net,name='fcout',num_hidden=out_dim)`
`    net = mx.sym.Activation(data=net,name='acout',act_type='sigmoid')`
`    net = mx.sym.Custom(data=net, label=labs, name='ce',op_type='CrossEntropyLoss')`
`    return net`

**Comment by eric-haibin-lin** [comment]
well, we cannot do anything about the FC layers since the first layer produces a dense matrix. What's the expectation?
